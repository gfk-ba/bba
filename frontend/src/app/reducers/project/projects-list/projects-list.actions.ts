import { ApiGetRequest } from '@dcs/ngx-tools';
import { projectsListManager } from './projects-list.manager';

export class FetchProjectsList extends ApiGetRequest {
  constructor() {
    super('projects', projectsListManager.actions.fetch.base, projectsListManager.schema);
  }
}
