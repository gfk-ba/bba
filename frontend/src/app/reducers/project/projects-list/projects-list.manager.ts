import { schema } from 'normalizr';
import { State } from '../../index';
import { currentProjectManager } from '../current-project/current-project.manager';
import { IProject, Project } from '../models/project.class';
import {
  crudCollectionManagerFactory,
  INormalizedCollectionState,
  generateNormalizedCollectionState,
} from '@dcs/ngx-tools';

export interface IProjectsListState extends INormalizedCollectionState {
  entities: { projects: { [key: string]: IProject } };
}

const initialState = generateNormalizedCollectionState<IProjectsListState>({ projects: {} });

const projectsSchema = new schema.Array(currentProjectManager.schema);

export const projectsListStateSelector = (state: State): IProjectsListState =>
  state.projects.projectsList;

export const projectsListManager = crudCollectionManagerFactory(
  'Projects List',
  'projects',
  initialState,
  currentProjectManager.actions,
  projectsListStateSelector,
  projectsSchema,
  Project
);
