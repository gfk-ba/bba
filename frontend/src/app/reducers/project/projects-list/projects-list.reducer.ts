import { Action } from '@ngrx/store';
import { IProjectsListState, projectsListManager } from './projects-list.manager';

export const projectsList = (
  state: IProjectsListState = projectsListManager.initialState,
  action: Action
): IProjectsListState => {
  switch (
    action.type
    // overwrite or add any actions here, just default redux
  ) {
  }

  return projectsListManager.reducer(state, action);
};
