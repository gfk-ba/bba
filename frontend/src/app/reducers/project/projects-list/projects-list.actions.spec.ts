import { FetchProjectsList } from './projects-list.actions';

describe('project list actions', () => {
  describe('FetchProjectsList', () => {
    it('builds the correct API request action', () => {
      expect(new FetchProjectsList()).toMatchSnapshot();
    });
  });
});
