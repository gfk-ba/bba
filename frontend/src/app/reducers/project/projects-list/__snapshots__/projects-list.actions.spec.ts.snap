// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project list actions FetchProjectsList builds the correct API request action 1`] = `
FetchProjectsList {
  "payload": Object {
    "cancel": undefined,
    "handlers": "[Projects List] Fetch",
    "normalizrSchema": ArraySchema {
      "schema": EntitySchema {
        "_getId": [Function],
        "_idAttribute": [Function],
        "_key": "projects",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
    },
    "rawDataProcessor": undefined,
    "request": Object {
      "method": "GET",
      "url": "projects",
    },
  },
  "type": "[ApiRequest] Execute API Request",
}
`;
